### DISCUSSION TITLE
How to remove Edges from tool palette

### DISCUSSION THREAD

**Discussion by wss29** [eclipse-glsp/glsp#948] [Q&A]
Hi,
    In my glsp application, there are no edges, I want to remove the edge label from tool palette, but I do not know how to do that, any help?
![image](https://user-images.githubusercontent.com/4537809/223660179-2cf5e62c-0dab-4ad0-9cbd-1bdc7aa36c8a.png)

**Comment by tortmayr** [comment]
HI @wss29,
the too palette is populated from the glsp-server using a `ToolPaletteItemProvider`. If you want to remove the edge section entirely you could override the `getItems` method of the [default implementation](https://github.com/eclipse-glsp/glsp-server-node/blob/v1.0.0/packages/server-node/src/features/contextactions/tool-palette-item-provider.ts#L66) e.g:
```ts
@injectable()
export class MyToolPaletteItemProvider extends DefaultToolPaletteItemProvider {
    override getItems(_args?: Args | undefined): PaletteItem[] {
        const handlers = this.operationHandlerRegistry.getAll().filter(CreateOperationHandler.is);
        this.counter = 0;
        const nodes = this.createPaletteItem(handlers, CreateNodeOperation.KIND);
        return [{ id: 'node-group', label: 'Nodes', actions: [], children: nodes, icon: 'symbol-property', sortString: 'A' }];
    }
}
```
and configure the new `ToolPaletteItemProvider` in your diagram module:
```ts
@injectable()
export class MyDiagramModule extends DiagramModule {
   ...
    protected override bindToolPaletteItemProvider(): BindingTarget<ToolPaletteItemProvider> | undefined {
        return MyToolPaletteItemProvider;
    }
}
  ```
